### ISSUE TITLE
Questions that share answer IDs don't work with satisfied by questions

### ISSUE THREAD

**Issue by bsnouffer** [shcaba/FishPath-updates#356]
If multiple questions in a group (satisfied by) question have the same ids for answers, then the answers for both questions will be treated as the same answer. For example, question A has answer ids 1, 2, 3 and question B has answer ids 1,2,3,4. Then trying to select question A answer 1 as a criteria will also select question B answer 1. 

Currently this is not causing any problems, but it could hinder updating or creating new grouped questions if using newer questions as part of them. I did encounter this when trying to update the fishery-dependent and independent questions in #324 and had to resort to editing the old questions instead of creating new questions as planned.

This happens because the logic assumes that answers will have unique ids across all questions. This was the case when the questions were input through an import (likely because ids were manually assigned?), but is not the case for questions that are added via the admin section. New questions always have answers starting with an ID of 2. See handleAddAnswer() in components\EditableAnswerList

Possible solutions:
Make all new answers have a unique ID. This would mean having to update all the old answers with overlapping IDs (or at least the ones that are needed for grouped questions)
Update grouped question logic to use both the question ID and answer ID to uniquely identify each answer. Would need to make sure that this is updated everywhere grouped questions are used and would also probably require re-entering or saving all the group questions so that they would contain the combo id instead of just answer id.

**Comment by shcaba** [comment]
Not considering the work that goes into it, option 2 seems the most future proof. Would you agree?

**Comment by bsnouffer** [comment]
Yes option 2 is likely more future proof. Either paradigm works, as long as you stick with the same assumption throughout the entire program. I think going back to fix it either way is going to a bit of a hassle to make it work with the data as currently saved. 
Given that we only know it affects creating new satisfied by questions at the moment, just documenting for now and likely won't get around to it for awhile unless something comes up in the peer review that requires this to be fixed. (I haven't seen it cause other issues, but would be good to note if seeing any other unexpected behavior from the grouped questions in results)
